### PR TITLE
Alfresco 23.x compatibility: instead of javax.servlet imports use jakarta.servlet

### PR DESCRIPTION
**Pull request by lethal1986**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,7 +5,6 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>net.bull.javamelody</groupId>
 	<artifactId>alfresco-javamelody-addon</artifactId>
-	<version>1.99.0</version>
```
Comment (evernat) [inline pom.xml:8]: why removing the version ? it can't work without a version

File: pom.xml [hunk 2]
```diff
@@ -34,7 +33,8 @@
 
 	<scm>
 		<connection>scm:git:git@github.com:javamelody/alfresco-javamelody.git</connection>
-		<developerConnection>scm:git:git@github.com:javamelody/alfresco-javamelody.git</developerConnection>
+		<developerConnection>
```
Comment (evernat) [inline pom.xml:36]: why reformating ?

File: pom.xml [hunk 3]
```diff
@@ -60,38 +60,54 @@
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-		<alfresco.version>5.2.g</alfresco.version>
```
Comment (evernat) [inline pom.xml:63]: why removing the alfresco.version ? it can't work without an alfresco.version used below

File: pom.xml [hunk 4]
```diff
@@ -60,38 +60,54 @@
...
 				<artifactId>alfresco-platform-distribution</artifactId>
 				<version>${alfresco.version}</version>
 				<type>pom</type>
 				<scope>import</scope>
+				<exclusions>
+					<exclusion>
+						<groupId>org.springframework.social</groupId>
+						<artifactId>spring-social-facebook</artifactId>
+					</exclusion>
+				</exclusions>
```
Comment (evernat) [inline pom.xml:104]: why excluding that ?

File: pom.xml [hunk 5]
```diff
@@ -163,5 +191,24 @@
...
+						<groupId>org.apache.maven.plugins</groupId>
+						<artifactId>maven-compiler-plugin</artifactId>
+						<configuration>
+							<source>17</source>
+							<target>17</target>
+						</configuration>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
```
Comment (evernat) [inline pom.xml:211]: why adding a profile which changes nothing ?

**Comment by evernat** [comment]
This PR can't be merged for reasons above.
https://github.com/javamelody/alfresco-javamelody/commit/0305b9f2b1ab84d97615ffea67f5b2a5e052f010 is better and merged for the next 2.3 release.

File: pom.xml (hunk 2 above)
Comment (lethal1986) [reply pom.xml:36]: my default IDE formatting rule for XML files...

File: pom.xml (hunk 3 above)
Comment (lethal1986) [reply pom.xml:63]: Do you think it is possible to update alfresco.version e.g. to 23.2?

File: pom.xml (hunk 4 above)
Comment (lethal1986) [reply pom.xml:104]: After upgrading alfresco.version from 5.2.g I was able to compile the maven project with only  this exclusion.

File: pom.xml (hunk 5 above)
Comment (lethal1986) [reply pom.xml:211]: Indeed. It is an unnecessary extension of pom file.

**Comment by lethal1986** [comment]
Hi evernat!
Indeed, pom.xml was incomplete and contained some unnecessary changes. Thanks for merging suggested changes, waiting for new release 2.3.

Best regards.

File: pom.xml (hunk 3 above)
Comment (evernat) [reply pom.xml:63]: @lethal1986 This alfresco.version is used for the version of alfresco-platform-distribution in the pom.xml below. There is no recent version of this artifact. Alfresco sdk may be updated first to 4.x (in-process sdk) and perhaps alfresco-platform-distribution should be replaced in the pom.xml by alfresco-content-services-community-distribution (ACS 6 and then ACS 23).
See https://docs.alfresco.com/content-services/latest/develop/sdk/
I see that there are some changes with sdk 4.x and that https://github.com/Alfresco/alfresco-sdk-samples/ has not changed in 9 years unfortunately.

Anyway, all these changes seem not necessary at the moment to build the plugin and it works already in Alfresco 23 after the commit https://github.com/javamelody/alfresco-javamelody/commit/0305b9f2b1ab84d97615ffea67f5b2a5e052f010

If you can suggest a new pull request for the SDK and/or ACS migration and if you can explain why you are doing the changes in the new PR, we may upgrade that version.

Comment (lethal1986) [reply pom.xml:63]: @evernat I have created a new pull request that uses SDK 4.9. In pom.xml now you can use 23.x in <alfresco.version>. 

Although - at the moment - I do not see any additional value for this update, your previous commit works well with Alfresco 23. SoI'll leave it up to you to merge my pull request into your code, or drop it.

Thanks and best regards.

**Pull request closed without merging**